src/encoding/byte_stream_split/decoder.rs

```diff
@@ -0,0 +1,61 @@
+use std::marker::PhantomData;
+use crate::error::Error;
+use crate::types::NativeType;
+
+/// Decodes according to [Byte Stream Split](https://github.com/apache/parquet-format/blob/master/Encodings.md#byte-stream-split-byte_stream_split--9).
+/// # Implementation
+/// This struct does not allocate on the heap.
+#[derive(Debug)]
+pub struct Decoder<'a, T: NativeType> {
+    values: &'a [u8],
+    buffer: Vec<u8>,
```

Comment:
I don't think we need the extra copy to `buffer`. We could directly `slice` from `values`.